core/gather/gatherers/anchor-elements.js

```diff
@@ -93,6 +93,16 @@ function collectAnchorElements() {
   // @ts-expect-error - put into scope via stringification
   const anchorElements = getElementsInDocument('a'); // eslint-disable-line no-undef
 
+  // Check, if document has only one lang attribute in opening html tag. If so,
+  // there is no need to run the `getLangOfInnerText()` function with multiple
+  // possible DOM traversals
+  /** @type {Array<HTMLElement|SVGElement>} */
+  // @ts-expect-error - put into scope via stringification
+  const langElements = getElementsInDocument('[lang]'); // eslint-disable-line no-undef
```

Comment:
Could this be more clear by instead explicitly checking `document.documentElement.lang`, and searching for further codes at `body [lang]`?